### PR TITLE
feat: [FC-8] config web hook push notification

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

File: src/chat/ChatScreen.tsx [hunk 1]
```diff
@@ -111,8 +116,18 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
       );
 
       await firebaseInstance.sendMessage(messages.text);
+
+      timeoutMessageRef.current = setTimeout(() => {
+        sendNotificationChat?.();
+      }, timeoutSendNotify);
```
Comment (danhmaisgt) [inline src/chat/ChatScreen.tsx:122]: I don't get why we have to wait for a while before sending push noti for that message but not instantly. Am I missing any requirements?

File: src/chat/ChatScreen.tsx [hunk 2]
```diff
@@ -134,9 +149,36 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
...
       firebaseInstance.clearConversationInfo();
+      if (timeoutMessageRef.current) {
+        clearTimeout(timeoutMessageRef.current);
+      }
     };
   }, [firebaseInstance]);
 
+  useEffect(() => {
+    let userConversation: () => void;
+    userConversation = firebaseInstance.userConversationListener(
```
Comment (honghoangsts) [inline src/chat/ChatScreen.tsx:160]: duplicate listener

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/chat/ChatScreen.tsx (hunk 2 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: yes, I removed it

File: src/chat/ChatScreen.tsx (hunk 1 above)
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx:149]: this help project when apply this package can send notification by used API. When user not in Chatscreen will get notification

File: src/chat/ChatScreen.tsx [hunk 3]
```diff
@@ -34,6 +34,8 @@ interface ChatScreenProps extends GiftedChatProps {
   maxPageSize?: number;
   inputToolbarProps?: IInputToolbar;
   hasCamera?: boolean;
+  sendNotificationChat?: () => void;
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: yes, I changed it

File: src/chat/ChatScreen.tsx (hunk 1 above)
Comment (danhmaisgt) [reply src/chat/ChatScreen.tsx:149]: I guess you forgot to cancel the push notification timeout when partners read the message within `timeoutSendNotify`. Please have a check @huydosgtech

File: src/chat/ChatScreen.tsx [hunk 4]
```diff
@@ -161,6 +178,9 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
     return () => {
       firebaseInstance.clearConversationInfo();
       chatDispatch?.(clearConversation());
+      if (timeoutMessageRef.current) {
+        clearTimeout(timeoutMessageRef.current);
+      }
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: yes, you're right. I removed

**Review by honghoangsts — APPROVED** [review]

**Review by hoangSTS — APPROVED** [review]

**Pull request merged**